### DISCUSSION TITLE
How to fix UnicodeDecodeError？

### DISCUSSION THREAD

**Discussion by VoyageHSSS** [chaidiscovery/chai-lab#91] [Q&A]
Could I ask the community for help on how to resolve this issue that occurs when I run python examples/predict_structure.py? Thank you!
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x80 in position 64: invalid start byte

![20240923123340](https://github.com/user-attachments/assets/b2b1e54f-07af-45b3-aff5-37b1019f21c1)
![20240923124144-1](https://github.com/user-attachments/assets/d00b8501-6a1e-4684-be8b-4b70f495c047)

![20240923124159-2](https://github.com/user-attachments/assets/c0a93a95-be54-4b1e-b9b1-77c8a09e59d1)

**Comment by arogozhnikov** [comment]
looks like you have non-ascii character somewhere in your input.

**Comment by VoyageHSSS** [comment]
> looks like you have non-ascii character somewhere in your input.

Thank you for your reply. The input file I am using is the default protein FASTA, retaining only the protein and small molecule, but the test fails. I also tried adding UTF-8 encoding in fasta_path.write_text(example_fasta), but I still encounter the same error.

**Comment by VoyageHSSS** [comment]
Additionally, I have changed the directory where the esm2 files are located and updated the path in esm2.py. However, I have only downloaded the single file esm2_t36_3B_UR50D.pt. Do I need to download more files? I am from China, and the default downloading method does not work for me.

**Comment by arogozhnikov** [comment]
> fasta_path.write_text(example_fasta)

first thing to check is `example_fasta.encode(encoding="ascii")`. It it fails, then the problem is in input.

> Do I need to download more files?

Yes, you need more files, see #29

**Comment by VoyageHSSS** [comment]
> > fasta_path.write_text(example_fasta)
> 
> first thing to check is `example_fasta.encode(encoding="ascii")`. It it fails, then the problem is in input.
> 
> > Do I need to download more files?
> 
> Yes, you need more files, see #29

Thank you for your guidance; I will try a bit more.
